src/protocol/commands/_vFile_readlink.rs

```diff
@@ -3,19 +3,23 @@ use super::prelude::*;
 #[derive(Debug)]
 pub struct vFileReadlink<'a> {
     pub filename: &'a [u8],
+
+    pub buf: &'a mut [u8],
 }
 
 impl<'a> ParseCommand<'a> for vFileReadlink<'a> {
     fn from_packet(buf: PacketBuf<'a>) -> Option<Self> {
-        let body = buf.into_body();
+        let (buf, body_range) = buf.into_raw_buf();
+        let (body, buf) = buf[body_range.start..].split_at_mut(body_range.end - body_range.start);
```

Comment:
Please refrain from using direct indexing operations from inside `gdbstub`, as it will introduce panicking code into the library. Unfortunately, AFAIK, there isn't a great way to automatically lint for this, so we must remain vigilant.

In this case, if I were writing this code, I'd probably end up using something like `cargo asm` to check the generated asm from this file, and/or throwing the code into godbolt.org and seeing what comes out.

That said, this is pretty tricky to do, and given that this code is in the packet parser for the `vFile` extension, it's not _super_ likely that it'll be used in a project with strict no panic properties...

TL;DR: Please add a comment to this code: `// TODO: rewrite to avoid panic`, and I'll get around to it myself at some point.